### ISSUE TITLE
Handling of site settings

### ISSUE THREAD

**Issue by sidneyjennings** [csyager/greeklink-core#37]
The site settings object can currently be deleted from the admin menu, which permanently disables site settings.

**Comment by csyager** [comment]
Yeah so the way SiteSettings is handled is pretty janky to say the least. It’s basically it’s own model which isn’t a really great way to do it but is the only way I could figure out to get something like dynamic environment variables to work. I reworked the admin module to treat it a little differently than other models, but theoretically yeah you could delete them. 

That should be ok though, because the getSettings() function at the top of views.py creates the SiteSettings object if it doesn’t already exist. So deleting it shouldn’t be an issue if that function is used properly throughout.

Adding a new one shouldn’t be a problem either but definitely won’t do what the user wants it to which is a usability problem. I think if we rework the admin module a bit that’ll pretty much solve this issue as best as possible. But yeah it’s an annoying system.

**Comment by csyager** [comment]
Ok, never mind I do see a problem.  Deleting the SiteSettings object made them uneditable later, because when a new one got created it gave it primary key equal to 2 instead of 1.  I've patched this temporarily, but yes I'll look into changing that admin page to make this less of an issue.
